app/models/katello/repository.rb

```diff
@@ -495,6 +495,24 @@ def ostree_branch_names
       self.ostree_branches.map(&:name)
     end
 
+    def update_ostree_branches!(branch_names)
+      non_uniq = branch_names.select { |e| branch_names.count(e) > 1 }.uniq
+      unless non_uniq.empty?
+        fail ::Katello::Errors::ConflictException,
+              _("Duplicate branches specified - %{branches}") % { branches: non_uniq.join(", ")}
+      end
+
+      # remove the ostree_branches not in this list
```

Comment:
Could consider a private function that encapsulates the behavior and description of the behavior instead of relying on a comment.